src/Sylius/Bundle/ThemeBundle/DependencyInjection/Compiler/ThemeCompilerPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class ThemeCompilerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->findDefinition('sylius.theme.repository');
+
+        $loader = $container->get('sylius.theme.loader');
+
+        $serializedThemes = [];
+        try {
+            $themeFiles = $container->get('sylius.theme.locator')->locate('theme.json', null, false);
+            foreach ($themeFiles as $themeFile) {
+                $serializedThemes[] = serialize($loader->load($themeFile));
+                $container->addResource(new FileResource($themeFile));
+            }
+
+            $definition->addArgument($serializedThemes);
```

Comment:
I would create method ``addSerializedThemesToDefinition`` or something like this, and throw whole "try" block content inside. I know it required some parameters ($definition and $loader, for example) but it is easier to read for someone who does not **want** to care about this adding themes logic. IMO it should be transparent, but of course I am a little bit pedantic ;)